Triangle.py

taniavill opened the pull request:
Hay dos funciones creadas que dibujan un triángulo, ninguna pasa los test, necesito ayuda!

tvillanueva committed:
Updated functions

lastone doesnt work

tvillanueva committed:
change print for return

tvillanueva committed:
updated

tvillanueva committed:
try

tvillanueva committed:
try 2

fernandezpablo85 committed:
Update pythonapp.yml

tvillanueva committed:
sum_all fixed

tvillanueva committed:
Merge branch 'master' of https://github.com/taniavill/python_init

tvillanueva committed:
Update number.py

All test enable, accept vararg and list as argument

tvillanueva committed:
sum_all arreglado y función de triángulo creada

mejora de number.sum_all  con funcion built in
la función tirangle.draw dibuja un triángulo pero no pasa la prueba, no encuentro el error, ayuda!

fernandezpablo85 commented on triangles/triangle_h_test.py:
```diff
@@ -3,7 +3,6 @@
-@pytest.mark.skip()
 @given(st.integers(min_value=1, max_value=100))
 def test_triangle_characters(n):
     t = triangle.draw(n=n)
```
tani, fijate que aca `t = triangle.draw`, o sea, se espera que la función `draw` devuelva el triángulo, que es lo que el test prueba.

vos creo que lo estás armando bien pero en vez de devolverlo (como un string con el triángulo) lo estás imprimiendo con `print` por eso no funciona

tvillanueva committed:
nueva función para dibujar triángulo

Tampoco pasa el test

tvillanueva committed:
funcion draw actualizada, pero aún no pasa las pruebas

tvillanueva committed:
función draw actualizada, dibuja triángulo llamando a otra función, no pasa test aún

seppo0010 commented on triangles/triangle.py:
```diff
@@ -1,19 +1,12 @@
+    for i in range (0,n):
+        triangulo.append(line(i+1))
+    print('\n'.join(triangulo))
+    #return '\n'.join(triangulo)
```
por qué print y no return? :cry:

taniavill replied:
Seppo, no me funciona con el return, me devuelve  '* \n* * \n* * * \n* * * * \n* * * * * '
Claro está que no pasa las pruebas, pablo me había sugerido usar el join para dibujar el triángulo, esta fue la única forma que encontré, si tenés alguna pista de como solucionarlo, o que puedo googlear, genial!

taniavill replied:
return print('\n'.join(triangulo)) si dibuja el triángulo pero tampoco pasa el test :/

seppo0010 commented on triangles/triangle.py:
```diff
@@ -1,19 +1,12 @@
+def line(x):
+    linea = []
+    for i in range(0,x):
+        linea.append('*')
```
`linea = ['*'] * x`

(en lo posible usá identificadores en inglés también)

tvillanueva committed:
Update triangle.py

se actualiza función y pasa los test. para dibujar el triángulo se tiene que llamar a la función con print

fernandezpablo85 commented on triangles/triangle.py:
```diff
@@ -1,11 +1,11 @@
-    """
-    # implement this.
-    return "TODO"
+def draw(n):
+    triangulo = []
+    for i in range (0,n):
+        triangulo.append(line(i+1))
+    return '\n'.join(triangulo)
+
+def line(x):
+    new_line = []
+    for i in range(0,x):
+        new_line.append('*')
+    return ' '.join(new_line)    
```
excelente Tani, te comparto como lo hice yo, muy parecido:

```python
def _line(n):
    stars = ["*" for _ in range(n)]
    return " ".join(stars)


def draw(n):
    lines = [_line(i + 1) for i in range(n)]
    return "\n".join(lines)
```

Algunos detalles:

- uso de [list comprehensions](https://realpython.com/list-comprehension-python/)
- usar `_line` en vez de `line` como nombre, para indicar que es una función accesoria de triángulo y que no está pensada para "el público general" (lo que en otros lenguajes de programación se consideraría "privada")

taniavill replied:
me re había olvidado de las lisas d comprensión y de las funciones privadas

Gracias!
